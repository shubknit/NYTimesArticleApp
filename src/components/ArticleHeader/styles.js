import {StyleSheet} from 'react-native';

/* Styling for Header Component */

export const styles = StyleSheet.create({
  headerStyle: {
    width: '100%',
    height: 80,
    backgroundColor: '#47E4C2',
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: 20,
    paddingRight: 20,
    justifyContent: 'space-between'
  },
  textStyle: {
    textAlign: 'center',
    color: '#fff',
    fontSize: 25
  }

})
