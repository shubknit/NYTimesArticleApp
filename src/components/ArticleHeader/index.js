import React from 'react';
import {View, Text} from 'react-native';
import {styles} from './styles';
import Icon from 'react-native-vector-icons/FontAwesome';

/* Header Component */

export const ArticleHeader = () => {
  return (
    <View style={styles.headerStyle}>
      <View style={styles.menu}>
        <Icon name="navicon" color="white" size={30}/>
      </View>
      <View>
        <Text style={styles.textStyle}>
          NyTimes Most Popular
        </Text>
      </View>
      <View style={styles.search}>
        <Icon name="search" color="white" size={30}/>
      </View>
    </View>
  )
}
