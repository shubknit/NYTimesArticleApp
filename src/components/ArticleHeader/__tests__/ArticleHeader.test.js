import React from 'react';
import { ArticleHeader } from '../../ArticleHeader';
import renderer from 'react-test-renderer';

/* Test for Header component */

it('should render the header  correctly', () => {
    const wrapper = renderer.create(<ArticleHeader/>).toJSON();
    expect(wrapper).toMatchSnapshot();
})