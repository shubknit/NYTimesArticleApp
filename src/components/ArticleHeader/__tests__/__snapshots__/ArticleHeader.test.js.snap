// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the header  correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#47E4C2",
      "flexDirection": "row",
      "height": 80,
      "justifyContent": "space-between",
      "paddingLeft": 20,
      "paddingRight": 20,
      "width": "100%",
    }
  }
>
  <View>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View>
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 25,
          "textAlign": "center",
        }
      }
    >
      NyTimes Most Popular
    </Text>
  </View>
  <View>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
